docs/conf.py

```diff
@@ -13,21 +13,65 @@
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
 import os
-import sys
+# import sys
 # sys.path.insert(0, os.path.abspath('.'))
+import re
+from functools import partial
+from pathlib import Path
+from typing import Callable, Match, Tuple, Union
+
+
+# -- Version forge -----------------------------------------------------------
+def _semver_matcher() -> Callable:
+    """Return a func performs regex matching for semantic versions.
+       The regex uses word_boundry and is sensitive to the changes
+       to the contents of the CMakeLists.txt file. The match should
+       have 3 matched groups:
+       1. `SET(TI_VERSION_MAJOR `
+       2. the corresponding version number
+       3. `)`
+    """
+    return lambda s, w: re.search(rf"(\bSET\(TI_VERSION_{w} )(\d+)(\b\))", s)
+
+
+def _intify_version(v: Match) -> int:
+    """Convert matched group v to integer."""
+    return int(v.group(2))
+
+
+def parse_semver(
+    cmakelist_path: str,
+    return_match_groups: bool = False
+) -> Union[Tuple[int, int, int], Tuple[Match[str], Match[str], Match[str]]]:
+    """Parse and return the major, minor and patch version numbers
+       (or matched groups) from CMakeLists.txt given CMAKELIST_PATH.
+    """
+    with open(cmakelist_path, "r") as fp:
+        cmakelist = fp.read()
+    matcher = partial(_semver_matcher(), cmakelist)
+    major, minor, patch = map(matcher, ["MAJOR", "MINOR", "PATCH"])
+    if return_match_groups:
+        return major, minor, patch
+    return tuple(map(_intify_version, (major, minor, patch)))
+
+
+# CMakeLists.txt is the only source of the truth when forging
+# the version, this script always reads from it, parses the version
+# and dump the version string to `version` file.
+cmake_file = Path(__file__).resolve().parents[1].joinpath('CMakeLists.txt')
+major, minor, patch = parse_semver(cmakelist_path=str(cmake_file))
+version_file = Path(__file__).resolve().parent.joinpath("version")
+taichi_version = f"{major}.{minor}.{patch}"
+with version_file.open('w') as f:
+    f.write(f"{taichi_version}\n")
+    print('Building doc version', taichi_version)
```

Comment:
re whether this should be included, please see the discussion section of the PR description.

Reply:
I think we can simply get rid of `docs/version` :-)